source/common/http/http1/codec_impl.cc

```diff
@@ -491,35 +499,41 @@ void ConnectionImpl::dispatch(Buffer::Instance& data) {
   // If an upgrade has been handled and there is body data or early upgrade
   // payload to send on, send it on.
   maybeDirectDispatch(data);
+  return ProtobufUtil::Status();
 }
 
-size_t ConnectionImpl::dispatchSlice(const char* slice, size_t len) {
+ProtobufUtil::StatusOr<ssize_t> ConnectionImpl::dispatchSlice(const char* slice, size_t len) {
   ssize_t rc = http_parser_execute(&parser_, &settings_, slice, len);
   if (HTTP_PARSER_ERRNO(&parser_) != HPE_OK && HTTP_PARSER_ERRNO(&parser_) != HPE_PAUSED) {
     sendProtocolError(Http1ResponseCodeDetails::get().HttpCodecError);
-    throw CodecProtocolException("http/1.1 protocol error: " +
-                                 std::string(http_errno_name(HTTP_PARSER_ERRNO(&parser_))));
+    return ProtobufUtil::Status(ProtobufUtil::error::Code::UNKNOWN,
```

Comment:
I'm unsure we will be able to use canonical error space of Status, as it would be better to have better error code than Unknown. However I do not know how to create custom error spaces, I know it was possible internally at Google. Need to look into this.